src/components/calcite-input/calcite-input.tsx

```diff
@@ -183,27 +240,270 @@ export class CalciteInput {
+
+  /**
+   * @internal
+   */
+  @Event() calciteInputFocus: EventEmitter;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInputBlur: EventEmitter;
+
+  /**
+   * This event fires when the value of the input changes.
+   */
+  @Event({ eventName: "calciteInputInput", cancelable: true }) calciteInputInput: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("keydown")
+  keyDownHandler(event: KeyboardEvent): void {
+    if (this.isClearable && getKey(event.key) === "Escape") {
+      this.clearInputValue(event);
+      event.preventDefault();
+    }
   }
 
-  get isClearable(): boolean {
-    return !this.isTextarea && (this.clearable || this.type === "search") && this.value.length > 0;
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** focus the rendered child element */
+  @Method()
+  async setFocus(): Promise<void> {
+    if (this.shouldFormatNumberByLocale()) {
+      this.childNumberEl?.focus();
+    } else {
+      this.childEl?.focus();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private clearInputValue = (event: KeyboardEvent | MouseEvent): void => {
+    const calciteInputInputEvent = this.emitInputFromUserInteraction("", event);
+    if (!calciteInputInputEvent.defaultPrevented) {
+      this.setValue("");
+    }
+  };
+
+  private decrementNumberValue = (nativeEvent: KeyboardEvent | MouseEvent): void => {
+    if (this.type === "number") {
+      const inputMin = this.minString ? parseFloat(this.minString) : null;
+      const inputStep = this.step === "any" ? 1 : Math.abs(this.step);
+      let inputVal = this.value && this.value !== "" ? parseFloat(this.value) : 0;
+      const decimals = this.value?.split(".")[1]?.length || 0;
+
+      let newValue = this.value;
+
+      if ((!inputMin && inputMin !== 0) || inputVal > inputMin) {
+        newValue = (inputVal -= inputStep).toFixed(decimals).toString();
+      }
+
+      const calciteInputInputEvent = this.emitInputFromUserInteraction(newValue, nativeEvent);
+
+      if (!calciteInputInputEvent.defaultPrevented) {
+        this.setValue(newValue);
+      }
+    }
+  };
+
+  private emitInputFromUserInteraction = (
+    value: string,
+    nativeEvent?: InputEvent | KeyboardEvent | MouseEvent
+  ): CustomEvent => {
+    return this.calciteInputInput.emit({
+      element: this.childEl,
+      nativeEvent,
+      value
+    });
+  };
+
+  private incrementNumberValue = (nativeEvent: KeyboardEvent | MouseEvent): void => {
+    if (this.type === "number") {
```

Comment:
Consolidate the increment/decrement value methods. There's a good chunk of reusable code between them.